include/seqan3/range/view/kmer_hash.hpp

```diff
@@ -12,47 +12,163 @@
+    
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    kmer_hash_view()                                       = default; //!< Defaulted.
+    kmer_hash_view(kmer_hash_view const & rhs)             = default; //!< Defaulted.
+    kmer_hash_view(kmer_hash_view && rhs)                  = default; //!< Defaulted.
+    kmer_hash_view & operator=(kmer_hash_view const & rhs) = default; //!< Defaulted.
+    kmer_hash_view & operator=(kmer_hash_view && rhs)      = default; //!< Defaulted.
+    ~kmer_hash_view()                                      = default; //!< Defaulted.
+    
+    //!\brief Construct from a view and a given shape.
+    kmer_hash_view(urng_t urange_, shape const s_) : urange{std::move(urange_)}, shape_(s_)
+    {}
+        
+    //!\brief Construct from a non-view that can be view-wrapped and a given shape.
+    template <std::ranges::ViewableRange orng_t>
+    kmer_hash_view(orng_t && urange_, shape const s_) : 
+        urange{std::view::all(std::forward<orng_t>(urange_))}, shape_(s_)
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the range.
+     * \returns Iterator to the first element.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto begin() noexcept
+    {
+        return iterator{std::ranges::begin(urange), shape_};
+    }
+    
+    //!\copydoc begin()
+    auto begin() const noexcept
+    {
+        return const_iterator{std::ranges::begin(urange), shape_};
+    }
+    
+    //!\copydoc begin()
+    auto cbegin() const noexcept
+    {
+        return const_iterator{std::ranges::begin(urange), shape_};
+    }
+    
+    /*!\brief Returns an iterator to the element following the last element of the range.
+     * \returns Iterator to the end.
+     *
+     * \details
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    auto end() noexcept
+    {
+        return std::ranges::end(urange);
+    }
+    
+    //!\copydoc end()
+    auto end() const noexcept
+    {
+        return std::ranges::end(urange);
+    }
+    
+    //!\copydoc end()
+    auto cend() const noexcept
+    {
+        return std::ranges::end(urange);
+    }
+    //!\}
+};
+
+//!\brief A deduction guide for the view class template.
+template <std::ranges::ViewableRange orng_t>
+kmer_hash_view(orng_t &&, shape const shape_) -> kmer_hash_view<std::ranges::all_view<orng_t>>;
+
+// ---------------------------------------------------------------------------------------------------------------------
+// kmer_hash_fn (adaptor definition)
+// ---------------------------------------------------------------------------------------------------------------------
 
 //![adaptor_def]
```

Comment:
?

The `//![adaptor_def]` ? Some documentation relies on this as an example.

Reply:
Please check whether that documentation is still accurately making use of this example now that you changed it completely :wink:

Reply:
resolved?

Reply:
The place where this is included, reads :

> This is the full proto-adaptor, first look at the second member function: it handles range and argument input and returns a range (in other cases this would delegate to the view's constructor if you have your own view type declared).

This needs to be adapted, or not?

Reply:
change to:

> This is the full proto-adaptor, first look at the second member function: it handles range and argument input and delegates to the view's constructor. In other, simpler cases you could invoke other adaptors here.